### DISCUSSION TITLE
proposing r.jina.ai as an alternative to FireCrawl to extract web content

### DISCUSSION THREAD

**Discussion by JOduMonT** [langgenius/dify#7803] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I'm using r.jina.ai to extract web content and it would be nice if it was proposed in DiFy.
Since r.jina.ai is already in the tools, I believe it would not be so complex to do.

### 2. Additional context or comments

I'm using r.jina.ai to extract web content and it would be nice if it was proposed in DiFy.
Since r.jina.ai is already in the tools, I believe it would be not so complex to do.

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by JOduMonT** [comment]
side comment; as a non Chinese user I don't understand why I'm force to click this
> [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）

**Reply by taowang1993** [reply]
Because some Chinese users will submit in Chinese, and Dify team is trying to keep the Discussion in English.

With regard to Jina, is it able to scrape subpages?

**Comment by Easonliuliang** [comment]
Another option worth considering: Purify (https://github.com/Easonliuliang/purify)

Single Go binary, no Redis/Playwright dependencies. Focuses on token reduction — tested 52-99% savings depending on the site. Also has a built-in MCP server if you are using Claude or Cursor.

Self-hosting is just one binary or docker compose up. Might be a good fit for Dify knowledge base ingestion where you want clean content without the overhead.
